### PR TITLE
Update padding of ragged features to enable dataloader change 

### PR DESCRIPTION
**Pull request by oliverholworthy**
### Goals :soccer:

- Enable a change to the merlin dataloader (https://github.com/NVIDIA-Merlin/dataloader/pull/103) that removes sparse output type and padding based on the value count property.
- Implement as a backwards compatible change for the current version as well as the future version  https://github.com/NVIDIA-Merlin/dataloader/pull/101

### Implementation Details :construction:

- Implements the equivalent of the padding currently in the dataloader
  - Uses ragged -> sparse -> dense conversion which appears to be faster than an alternative approach assigning values to a tensor constructed with zeros 

### Testing Details :mag:

- Existing tests should cover existing usage of the Merlin Dataloader
- Adds unit tests for padding function to check pad batch

### CODE REVIEW

File: transformers4rec/torch/utils/data_utils.py [hunk 1]
```diff
@@ -367,6 +368,34 @@ def __init__(
...
+        return pad_fn
+
+    @staticmethod
+    def _augment_schema(
+        schema,
+        cats=None,
+        conts=None,
+        labels=None,
+    ):
+        schema = schema.select_by_name(conts + cats + labels)
```
Comment (marcromeyn) [inline transformers4rec/torch/utils/data_utils.py:387]: Maybe we should remove the `=None` for cats, conts & labels?

File: transformers4rec/torch/utils/padding.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    ------
+    ValueError
+        If ragged column found with no padding length provided
+    """
+    if batch is None or not isinstance(batch, dict):
+        return batch
+
+    batch_padded = {}
+    for k, values in batch.items():
+        if k.endswith("__values"):
```
Comment (marcromeyn) [inline transformers4rec/torch/utils/padding.py:84]: Wouldn't it be better to put `__values` and `__offsets` as constants somewhere in merlin-core?

File: transformers4rec/torch/utils/data_utils.py [hunk 3]
```diff
@@ -367,6 +368,34 @@ def __init__(
...
+    def _get_pad_fn(padding_lengths):
+        def pad_fn(x, y):
+            new_x = pad_batch(x, padding_lengths)
+            new_y = pad_batch(y, padding_lengths)
+            return new_x, new_y
+
+        return pad_fn
+
+    @staticmethod
+    def _augment_schema(
```
Comment (oliverholworthy) [inline transformers4rec/torch/utils/data_utils.py:381]: Moved this from `schema_utiils` to here so that it's closer to the only place it's called in the codebase.

File: transformers4rec/torch/utils/data_utils.py (hunk 1 above)
Comment (oliverholworthy) [reply transformers4rec/torch/utils/data_utils.py:387]: that's clearer without the default now. it wouldn't have worked as None before this change either.

Comment (oliverholworthy) [reply transformers4rec/torch/utils/data_utils.py:391]: On second thought, I've put the None back and added support for the None to the method by setting to an empty list if not provided. 

It doesn't appear to be captured by any tests, but this would enable training a model with only continuous or only categorical features. Currently you need to have at least one of each for the current version of this MerlinDataloader to work.

**Review by karlhigley — APPROVED** [review]
